### ISSUE TITLE
BigDecimal support

### ISSUE THREAD

**Issue by avently** [cbeust/klaxon#117]
BigInteger is supported but BigDecimal isn't. Can it be supported too?

**Comment by cbeust** [comment]
Can you give an example of a JSON document and the kind of parsing you'd expect?

**Comment by avently** [comment]
```
data class A(val data: BigDecimal)
        val something = Klaxon().parse<A>("""
            {"data": "0.00000001"}
            """)
        println(something)
```

```Couldn't find a suitable constructor for class A to initialize with {data=0.00000001}: java.lang.IllegalArgumentException argument type mismatch```

I have a class with a bunch of vars. Some of them has type BigDecimal. I'm writting all class data to JSON file and retrieving it sometimes (it's a class for settings).

Also i'm getting such exception in my other class and i don't really know what field is causing this. Could the error message be more verbose? For example, argument type mismatch: "field 'data' cannot be parsed"

**Comment by cbeust** [comment]
Fixed in 2.1.8.

**Comment by avently** [comment]
Thank you, Cedric
